api/Controllers/HotelsController.cs

```diff
@@ -43,13 +47,35 @@ public async Task<IActionResult> CreateHotel([FromBody] AddHotelRepresentation r
             return CreatedAtAction(nameof(GetHotel), new { id = hotelEntity.Id }, null);
         }
 
+        [Route("{id}")]
+        [HttpPut]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> EditHotel([FromRoute] int id, [FromBody] EditHotel request)
+        {
+            if (request == null)
+            {
+                return BadRequest("No data to change hotel");
+            }
+            try
+            {
+                HotelEntity hotelEntity = await _hotelsDb.Hotels.SingleOrDefaultAsync(h => h.Id == id);
+                _mapper.Map(request, hotelEntity);
+                await _hotelsDb.SaveChangesAsync();
+                return NoContent();
+            }
```

Comment:
Do you really want to use try/catch? Any ideas how to avoid? Is it needed to avoid try/catch? What can cause exception there?

Reply:
I should check each possible exception using "if"